sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -220,3 +222,80 @@ def test_singular_values():
     rpca.fit(X_hat_rpca)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+
+
+def test_svd_transform_fit_transform():
+    for whiten in (False, True):
+        tsvd = TruncatedSVD(n_components=X.shape[1] - 1,
+                            random_state=42, whiten=True)
+        Xt = tsvd.fit_transform(X)
+        Xt_2 = tsvd.transform(X)
+        assert_allclose(Xt, Xt_2)
+
+
+def test_svd_consistency():
+    X_c = X - X.mean(axis=0)
+    svd = TruncatedSVD(n_components=5, whiten=True, random_state=0)
+    assert_allclose(svd.fit(X_c).explained_variance_,
+                    np.ones(5))
+
+
+def test_truncated_svd_eq_pca():
+    # TruncatedSVD should be equal to PCA on centered data
+
+    X_c = X - X.mean(axis=0)
+
+    params = dict(n_components=10, random_state=42)
+
+    for whiten in (False, True):
+        svd = TruncatedSVD(whiten=whiten, algorithm='arpack', **params)
+        pca = PCA(whiten=whiten, svd_solver='arpack',
+                  **params)
+
+        Xt_svd = svd.fit_transform(X_c)
+        Xt_pca = pca.fit_transform(X_c)
+
+        if whiten is True:
+            rtol = 9e-3
+        else:
+            rtol = 1e-9
+        assert_allclose(Xt_svd, Xt_pca, rtol=rtol)
+
+        assert_allclose(pca.mean_, 0, atol=1e-9)
+
+        assert_allclose(svd.components_, pca.components_)
+
+        # transform only a subset of the data
+        Xt2_svd = svd.transform(X_c[:10])
+        Xt2_pca = pca.transform(X_c[:10])
+        assert_allclose(Xt2_svd, Xt2_pca, rtol=rtol)
+
+        X2_svd = svd.inverse_transform(Xt2_svd)
+        X2_pca = pca.inverse_transform(Xt2_svd)
+        assert_allclose(X2_svd, X2_pca, rtol=rtol)
+
+
+def test_lsi():
+    # Verification of the Latent Semantic Indexing example from
+    # Grossman, Frieder (2004) Information Retrieval, Algorithms and Heuristics
+    # http://www1.se.cuhk.edu.hk/~seem5680/lecture/LSI-Eg.pdf
+    from sklearn.feature_extraction.text import CountVectorizer
+    from sklearn.metrics.pairwise import cosine_similarity
+    documents = ["Shipment of gold damaged in aa fire.",
```

Comment:
warrants a small comment explaining why we must use aa instead of a for future devs